### PR TITLE
[4.0] Featured icon display

### PR DESCRIPTION
**Pull request by brianteeman**
If you have a user who does **not** have core.edit.state permissions for com_content.article then the icon displayed for "featured/unfeatured" is wrong and you cannot differentiate between  "featured/unfeatured" . 

The problem is that the style comes from the `tbody-icon` on the button and the code in the joomla.icons.action-button layout has two bugs.

The first bug is that **disabled** was in the class attribute - it should be its own attribute. This meant that the button was never disabled. Which is probably why the second bug was introduced - the entire `if(!empty($disabled))` test is not required and because it is not wrapped in a button it never gets the `tbody-icon` class and looks **wrong**

### Testing instructions
Create a new user at the manager level. In article options change the permissions for manager so that **Edit State** is disabled.

Login as that user and go to the list of articles.

### Before PR
image

### After PR
image

### CODE REVIEW

File: layouts/joomla/button/action-button.php [hunk 1]
```diff
@@ -25,16 +25,12 @@
...
-			onclick="return Joomla.listItemTask('<?php echo $checkboxName . $this->escape($row ?? ''); ?>', '<?php echo $this->escape(isset($task) ? $taskPrefix . $task : ''); ?>')"
-		<?php endif; ?>
-	>
-		<span class="<?php echo $this->escape($icon ?? ''); ?>" aria-hidden="true"></span>
-		<span class="sr-only"><?php echo Text::_($title); ?></span>
-	</button>
-<?php endif; ?>
+<button type="submit" class="tbody-icon data-state-<?php echo $this->escape($value ?? ''); ?>" 
+	<?php echo $this->escape(!empty($disabled) ? 'disabled' : null); ?>
+	<?php if(!empty($task) && empty($disabled)): ?>
```
Comment (Quy) [inline layouts/joomla/button/action-button.php]:
```suggestion
	<?php if (!empty($task) && empty($disabled)) : ?>
```

File: layouts/joomla/button/action-button.php [hunk 2]
```diff
@@ -25,16 +25,12 @@
...
-		<?php if(!empty($task) && empty($disabled)): ?>
-			onclick="return Joomla.listItemTask('<?php echo $checkboxName . $this->escape($row ?? ''); ?>', '<?php echo $this->escape(isset($task) ? $taskPrefix . $task : ''); ?>')"
-		<?php endif; ?>
-	>
-		<span class="<?php echo $this->escape($icon ?? ''); ?>" aria-hidden="true"></span>
-		<span class="sr-only"><?php echo Text::_($title); ?></span>
-	</button>
-<?php endif; ?>
+<button type="submit" class="tbody-icon data-state-<?php echo $this->escape($value ?? ''); ?>" 
+	<?php echo $this->escape(!empty($disabled) ? 'disabled' : null); ?>
```
Comment (Quy) [inline layouts/joomla/button/action-button.php]:
```suggestion
	<?php echo !empty($disabled) ? 'disabled' : ''; ?>
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on e2779df1219d9c99222b24debc7da5f37bc2570e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25400">issues.joomla.org/tracker/joomla-cms/25400</a>.</sub>

File: layouts/joomla/button/action-button.php [hunk 3]
```diff
@@ -25,16 +25,12 @@
...
-		<?php if(!empty($task) && empty($disabled)): ?>
-			onclick="return Joomla.listItemTask('<?php echo $checkboxName . $this->escape($row ?? ''); ?>', '<?php echo $this->escape(isset($task) ? $taskPrefix . $task : ''); ?>')"
-		<?php endif; ?>
-	>
-		<span class="<?php echo $this->escape($icon ?? ''); ?>" aria-hidden="true"></span>
-		<span class="sr-only"><?php echo Text::_($title); ?></span>
-	</button>
-<?php endif; ?>
+<button type="submit" class="tbody-icon data-state-<?php echo $this->escape($value ?? ''); ?>" 
+	<?php echo !empty($disabled) ? 'disabled' : ''; ?>
```
Comment (Quy) [inline layouts/joomla/button/action-button.php:29]: `$disabled` is a boolean so no need to test `empty` again per line 24. Same in line 30.

Comment (brianteeman) [reply layouts/joomla/button/action-button.php:29]: sorry i dont understand

Comment (Quy) [reply layouts/joomla/button/action-button.php:29]: Line 24: `$disabled = !empty($options['disabled']);`
Thus, `$disabled` is a boolean.
`!empty($disabled)` can just be `$disabled`.
`empty($disabled)` can just be  `!$disabled` in the next line.

This is probably beyond the scope of this PR. I can do a separate PR if you prefer.

Comment (brianteeman) [reply layouts/joomla/button/action-button.php:29]: please do a separate pr as I am not convinced

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e2779df1219d9c99222b24debc7da5f37bc2570e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25400">issues.joomla.org/tracker/joomla-cms/25400</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25400">issues.joomla.org/tracker/joomla-cms/25400</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

File: layouts/joomla/button/action-button.php (hunk 3 above)
Comment (wilsonge) [reply layouts/joomla/button/action-button.php:29]: I'm convinced. But like you both said separate PR is fine too

**Comment by brianteeman** [comment]
Thanks
